File: iOSbeautyOfDevelop/chapter4/4.3/lifeCycle/lifeCycle/ViewController.m
//
//  ViewController.m
//  lifeCycle
//
//  Created by aa64mac on 8/16/15.
//  Copyright (c) 2015 cc. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    NSLog(@"viewWillAppear, animated:%@", animated?@"Yes":@"No");
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    NSLog(@"viewDidAppear, animated:%@", animated?@"Yes":@"No");
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    NSLog(@"viewWillDisappear, animated:%@", animated?@"Yes":@"No");
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    NSLog(@"viewDidLoad");
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Start Developing iOS Apps (Swift)/CrashApp/CrashApp/ViewController.swift
//
//  ViewController.swift
//  CrashApp
//
//  Created by aa64mac on 10/6/15.
//  Copyright © 2015 cyj. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func crashTouchUpInside(sender: AnyObject) {
        print("It will crash.")
        let z = 0
        let x = devideFunc(z)
        print(x)
    }
    
    func devideFunc(z: Int)->Int {
        return 1/z
    }

}

